buildroot-external/patches/occu/0061-WebUI-Fix-SortFavoritesComboboxOnStartpageByName.patch

```diff
@@ -0,0 +1,10 @@
+--- occu/WebUI/www/rega/pages/startpage.htm.orig
++++ occu/WebUI/www/rega/pages/startpage.htm
+@@ -50,6 +50,7 @@
+             { 
+               Write( '<option>${noFav}</option>' ); 
+             }
++            oFavoriteList.SortByName();

```

Comment:
Anstatt das `SortBaName()` bei jedem Aufruf von `startpage.htm` aufzurufen sollte es IMHO effektiver/ressourcenschonender sein das SortByName besser in den Routinen aufzurufen nach de Editieren des Namens bzw. anlegen eines neuen Favoriten.

Reply:
Da sehe ich leider nicht so recht durch. 
Da werden die IDs umgeshiftet; der zuletzt editierte wandert nach ganz unten, nachdem vorher auch eine Kopie angelegt wurde.
Müsste wohl irgendwo hier passieren: [/www/rega/esp/favorites.fn#L233](https://github.com/eq-3/occu/blob/84fbd2d3783f2e827a02c13d85324d741ace9f2f/WebUI/www/rega/esp/favorites.fn#L233)
Ich schaue es mir die Tage noch mal an 👍

Reply:
Fazit nach einer Stunde intensiver Beschäftigung mit der `SaveFavoriteList` Methode: 
Ich blick da absolut nicht durch 😣 
Es mangelt schon am Verständnis undokumentierten verwendeten internen Methoden:
```
      integer iOrigID = oCopy.EnCopyID();
      object oOrig = dom.GetObject( iOrigID );
      if( oOrig )
      {
        oCopy.EnumCopyTo( oOrig.ID() );
        dom.DeleteObject( oCopy.ID() );
        oOrig.EnCopyID( ID_ERROR );
      }
```
Aber auch mit viel Debug-Output bekomme ich es nicht auf die Reihe, an irgendeiner Stelle die Sortierung einzubauen. 😞

Reply:
Also ich hab vorerst auch kapituliert. An den Stellen werden anscheinend Objekte hin/her-kopiert um irgendwie eine neue Favoritenliste zu erzeugen falls Objekte entfernt werden. Da schein deine ursprüngliche Änderung mit einfach SortByName() doch die effektivere zu sein.